derive-error-chain-tests/src/main.rs

```diff
@@ -269,6 +269,66 @@ mod attributes_test {
 	}
 }
 
+mod generics_test {
+	use std::{error, fmt, io};
+
+	mod inner1 {
+		use std::fmt;
+		#[derive(Debug, error_chain)]
+		pub enum ErrorKind<T: Send + fmt::Debug + 'static> {
+			Msg(String),
+
+			#[error_chain(custom)]
+			CustomGeneric(T)
+		}
+	}
+
+	mod inner2 {
+		#[derive(Debug, error_chain)]
+		pub enum ErrorKind {
+			Msg(String),
+		}
+	}
+
+	#[derive(Debug, error_chain)]
+	pub enum ErrorKind<T: error::Error + Send + 'static, U>
+		where U: Send + fmt::Debug + fmt::Display + 'static {
+		Msg(String),
+
+		#[error_chain(custom)]
+		#[error_chain(description = r#"|_| "custom error""#)]
+		#[error_chain(display = r#"|t| write!(f, "custom error: {}", t)"#)]
+		Custom(String),
+
+		#[error_chain(custom)]
+		#[error_chain(description = r#"|_| "custom generic error""#)]
+		#[error_chain(display = r#"|t| write!(f, "custom generic error: {}", t)"#)]
+		CustomGeneric(T),
+
+		#[error_chain(custom)]
+		#[error_chain(description = r#"|_| "custom generic boxed error""#)]
+		#[error_chain(display = r#"|t| write!(f, "custom generic boxed error: {}", t)"#)]
+		CustomGenericBoxed(Box<U>),
+
+		#[error_chain(link = "inner1::Error<U>")]
+		LinkGeneric(inner1::ErrorKind<U>),
+
+		#[error_chain(link = "inner2::Error")]
+		Link(inner2::ErrorKind),
+
+		// FIXME: conflicting implementations of trait `std::convert::From<&str>` for type `generics_test::Error<&str, _>
```

Comment:
Is this accurate? AFAICT this would only happen if `T` was a `String` or `&str`, but that can't be because it both violates the semantics of foreign errors and doesn't satisfy the `T: Error` generic bound anyway.